From 1050f7c83cde0c8ad7b41b3e95face9c99288d54 Mon Sep 17 00:00:00 2001
From: Felix Radensky <felix.r@variscite.com>
Date: Thu, 28 Oct 2021 09:58:32 -0700
Subject: [PATCH] usb: ehci-mx6: Fix USB type detection

When external connector is defined the USB type validy check fails.
Fix the check to take external connector into account.

(cherry picked from commit 0e4c5d6036cc1dabd48bfe3a01dc7167a4ccbab6)
---
 drivers/usb/host/ehci-mx6.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/ehci-mx6.c b/drivers/usb/host/ehci-mx6.c
index bc57630e38f..0095902d5e2 100644
--- a/drivers/usb/host/ehci-mx6.c
+++ b/drivers/usb/host/ehci-mx6.c
@@ -372,7 +372,7 @@ static int ehci_usb_of_to_plat(struct udevice *dev)
 	struct usb_plat *plat = dev_get_plat(dev);
 	struct ehci_mx6_priv_data *priv = dev_get_priv(dev);
 	enum usb_dr_mode dr_mode;
-	const struct fdt_property *extcon;
+	const struct fdt_property *extcon = NULL;
 
 	extcon = fdt_get_property(gd->fdt_blob, dev_of_offset(dev),
 			"extcon", NULL);
@@ -395,7 +395,7 @@ static int ehci_usb_of_to_plat(struct udevice *dev)
 	};
 
 check_type:
-	if (priv->init_type != USB_INIT_UNKNOWN && priv->init_type != plat->init_type) {
+	if (priv->init_type != USB_INIT_UNKNOWN && !extcon && priv->init_type != plat->init_type) {
 		debug("Request USB type is %u, board forced type is %u\n",
 			plat->init_type, priv->init_type);
 		return -ENODEV;
-- 
2.34.1

